src/Device/SmartDeviceSensors.cpp

```
@@ -91,6 +93,57 @@ DeviceDescriptor::OnLocationSensor(std::chrono::system_clock::time_point time,
+      snprintf(hdop_str, sizeof(hdop_str), "%.2f", accuracy);
+
+    if ((int)sizeof(sentence) > (4 + snprintf(sentence, sizeof(sentence),
+             "$GPRMC,%s,A,%s,%s,%s,%02d%02d%02d,,,A",
+             ts_str, loc_str, speed_str, cog_str,
+             date_time.day, date_time.month, date_time.year % 100))) {
+      AppendNMEAChecksum(sentence);
+      nmea_logger->Log(sentence);
+    }
+
+    if ((int)sizeof(sentence) > (4 + snprintf(sentence, sizeof(sentence),
+             "$GPGGA,%s,%s,1,%d,%s,%s,,",
```

Cppcheck (2.20.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)